### PR TITLE
Enforce BankInformation Currency and update default Instructions

### PR DESCRIPTION
**Pull request by kewitz**
Following the feedback received in our last demos.

### CODE REVIEW

File: components/edit-collective/sections/BankTransfer.js [hunk 1]
```diff
@@ -82,8 +88,21 @@ const BankTransfer = props => {
...
-  const useStructuredForm = !existingManualPaymentMethod || (existingManualPaymentMethod && bankAccount) ? true : false;
+  const existingPayoutMethod = data.host.payoutMethods.find(pm => pm.data.isManualBankTransfer);
+  const useStructuredForm =
+    !existingManualPaymentMethod || (existingManualPaymentMethod && existingPayoutMethod) ? true : false;
+  const instructions =
+    get(data.host, 'settings.paymentMethods.manual.instructions') || BANK_TRANSFER_DEFAULT_INSTRUCTIONS;
+  const fixedCurrency =
+    // Fix currency if the existing payout method already matches the collective currency
+    (!existingPayoutMethod || existingPayoutMethod.data.currency === data.host.currency) &&
+    // or if it was already defined by Stripe
```
Comment (znarf) [inline components/edit-collective/sections/BankTransfer.js:99]: The comment say "or" while the code say `&&`. It's difficult to follow what is being done here.

File: components/expenses/PayoutBankInformationForm.js [hunk 2]
```diff
@@ -219,27 +219,28 @@ const availableCurrenciesQuery = gqlV2`
...
     context: API_V2_CONTEXT,
     variables: { slug: host.slug },
-    skip: Boolean(host.transferwise?.availableCurrencies),
+    skip: Boolean(fixedCurrency || host.transferwise?.availableCurrencies),
   });
   const formik = useFormikContext();
   const { formatMessage } = useIntl();
-  const availableCurrencies = host.transferwise?.availableCurrencies || data?.host?.transferwise?.availableCurrencies;
+  const availableCurrencies =
+    fixedCurrency || host.transferwise?.availableCurrencies || data?.host?.transferwise?.availableCurrencies;
```
Comment (znarf) [inline components/expenses/PayoutBankInformationForm.js:231]: It's weird to put `fixedCurrency` in what was until now expected to be an array. We should generally avoid long chains of `||` or `&&` for variable assignment.

File: components/expenses/PayoutBankInformationForm.js [hunk 3]
```diff
@@ -219,27 +219,28 @@ const availableCurrenciesQuery = gqlV2`
...
+  const selectedCurrency = fixedCurrency || get(formik.values, currencyFieldName);
 
   if (!availableCurrencies) {
     return <StyledSpinner />;
   }
 
-  const currencies = formatStringOptions(availableCurrencies);
-  const currencyFieldName = getFieldName('data.currency');
-  const selectedCurrency = get(formik.values, currencyFieldName);
+  const currencies = formatStringOptions(fixedCurrency ? [fixedCurrency] : availableCurrencies);
```
Comment (znarf) [inline components/expenses/PayoutBankInformationForm.js:239]: Why not just read from `availableCurrencies`.

File: components/edit-collective/sections/BankTransfer.js (hunk 1 above)
Comment (znarf) [reply components/edit-collective/sections/BankTransfer.js:99]: For clarity, we should avoid long chains like that for assignment.

Comment (kewitz) [reply components/edit-collective/sections/BankTransfer.js:99]: Indeed! It lacks clarity and the logic can be simplified.

File: components/expenses/PayoutBankInformationForm.js (hunk 2 above)
Comment (kewitz) [reply components/expenses/PayoutBankInformationForm.js:231]: This logic can also be simplified.

**Pull request merged**